cmd/libs/go2idl/client-gen/generators/client_generator.go

```
@@ -279,3 +281,27 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 
 	return generator.Packages(packageList)
 }
+
+// tagOverrideNamer is a namer which pulls names from a given tag, if specified,
+// and otherwise falls back to a different namer.
+type tagOverrideNamer struct {
+	tagName  string
+	fallback namer.Namer
+}
+
+func (n *tagOverrideNamer) Name(t *types.Type) string {
+	if nameOverride := extractTag(n.tagName, t.SecondClosestCommentLines); nameOverride != "" {
+		return nameOverride
+	}
+
+	return n.fallback.Name(t)
```

slightly concerned about silent failures, should I be?